tests/integration/test_nvt_tf_training.py

```diff
@@ -0,0 +1,452 @@
+        cont_names=NUMERIC_COLUMNS,
+        engine="parquet",
+        shuffle=False,
+        buffer_size=0.06,
+        parts_per_chunk=1,
+    )
+
+    inputs = {}  # tf.keras.Input placeholders for each feature to be used
+    emb_layers = []  # output of all embedding layers, which will be concatenated
+
+    for col in CATEGORICAL_COLUMNS:
+        inputs[col] = tf.keras.Input(name=col, dtype=tf.int32, shape=(1,))
+    # Note that we need two input tensors for multi-hot categorical features
+    for col in CATEGORICAL_MH_COLUMNS:
+        inputs[col + "__values"] = tf.keras.Input(name=f"{col}__values", dtype=tf.int64, shape=(1,))
+        inputs[col + "__nnzs"] = tf.keras.Input(name=f"{col}__nnzs", dtype=tf.int64, shape=(1,))
+
+    for col in CATEGORICAL_COLUMNS + CATEGORICAL_MH_COLUMNS:
+        emb_layers.append(
+            tf.feature_column.embedding_column(
+                tf.feature_column.categorical_column_with_identity(
+                    col, EMBEDDING_TABLE_SHAPES[col][0]
+                ),  # Input dimension (vocab size)
+                EMBEDDING_TABLE_SHAPES[col][1],  # Embedding output dimension
+            )
+        )
+
+    emb_layer = layers.DenseFeatures(emb_layers)
+    x_emb_output = emb_layer(inputs)
+
+    x = tf.keras.layers.Dense(128, activation="relu")(x_emb_output)
+    x = tf.keras.layers.Dense(128, activation="relu")(x)
+    x = tf.keras.layers.Dense(128, activation="relu")(x)
+    x = tf.keras.layers.Dense(1, activation="sigmoid", name="output")(x)
+
+    model = tf.keras.Model(inputs=inputs, outputs=x)
+    model.compile("sgd", "binary_crossentropy")
+
+    validation_callback = KerasSequenceValidater(valid_dataset_tf)
+
+    model.fit(train_dataset_tf, callbacks=[validation_callback], epochs=1)
+
+    sample_data_trans = KerasSequenceLoader(
+        os.path.join(INPUT_DATA_DIR, "test_data_trans.parquet"),
+        batch_size=TEST_N_ROWS,
+        label_names=["rating"],
+        cat_names=CATEGORICAL_COLUMNS + CATEGORICAL_MH_COLUMNS,
+        cont_names=NUMERIC_COLUMNS,
+        engine="parquet",
+        shuffle=False,
+        buffer_size=0.06,
+        parts_per_chunk=1,
+    )
+
+    pred = model.predict(sample_data_trans)
+    cudf_pred = cudf.DataFrame(pred)
+    cudf_pred.to_csv(os.path.join(INPUT_DATA_DIR, "output.csv"))
+
+    MODEL_NAME_TF = os.environ.get("MODEL_NAME_TF", "movielens_tf")
+    MODEL_PATH_TEMP_TF = os.path.join(MODEL_BASE_DIR, MODEL_NAME_TF, "1/model.savedmodel")
+
+    model.save(MODEL_PATH_TEMP_TF)
+
+    workflow = nvt.Workflow.load(os.path.join(INPUT_DATA_DIR, "workflow"))
+
+    workflow.output_dtypes["userId"] = "int32"
+    workflow.output_dtypes["movieId"] = "int32"
+
+    MODEL_NAME_ENSEMBLE = "test_movielens_tf"
+    # model path to save the models
+    MODEL_PATH = os.path.join(MODEL_BASE_DIR, "models")
+
+    export_tensorflow_ensemble(model, workflow, MODEL_NAME_ENSEMBLE, MODEL_PATH, ["rating"])
+
+
+def test_nvt_tf_rossmann_training(tmpdir):
```

Comment:
How is this different than the train notebook?  We're already running the TF training script here 

https://github.com/NVIDIA/NVTabular/blob/9e14acc4eeb103bdd9590fb2a28eb5a182151be2/tests/integration/test_notebooks.py#L144-L146

as part of the integration tests.

Is this code just cut and paste from the TF notebook - with the only difference being that we export the model to triton at the end?  If so maybe we should just add the export_tensorflow_ensemble call to the end of the rossmann training notebook like we do for movielens?

Reply:
_Can you work with Alberto on this?_

Sure! That's a pretty good idea!

_Is this code just cut and paste from the TF notebook_ 

Most of them, Yes. I added extra code that runs NVT transform and TF predict to test the whole Triton Pipeline. This test will also help us to test the C++ backend.

If you think we add that test part in the notebooks, that works for me but I didn't think that'd be something useful for the users.

Reply:
_Also can you run black on your code?_

I actually did it. I am not sure why CI is giving this error. Maybe, I should update the black on my conda env.

Reply:
> I actually did it. I am not sure why CI is giving this error. Maybe, I should update the black on my conda env

We have a pre-commit hook configured with the right version of black on it https://pre-commit.com/ if you enable this it'll run automatically on every commit you make

Reply:
> If you think we can add that test part in the notebooks, that works for me but I didn't think that'd be something useful for the users.

I think we can keep the training in the notebooks (to reduce duplicate code) but can add the inference tests themselves separate from the notebooks